lib/api/objectPutACL.js

```diff
@@ -113,6 +113,11 @@ export default function objectPutACL(authInfo, request, log, cb) {
                             errors.NoSuchKey;
                         return next(err, bucket);
                     }
+                    if (objectMD.isDeleteMarker) {
+                        log.trace('delete marker detected',
```

Comment:
Yes. Although, I think it might make more sense if we return 404 NoSuchKey if someone tries to put an acl on an object whose latest version is a delete marker without specifying the version id.